assignments/burmanm/week-2.scm

```diff
@@ -0,0 +1,394 @@
+;; Procedures and the Processes They Generate (1.2)
+
+;; Exercise 1.9
+;; First one is recursive as we need to keep the stack since we're unable to calculate the
+;; value without knowing the subprocesses return value.
+;;
+;; (+ 4 5)
+;;   (inc (+ 3 5))
+;;     (inc (inc (+ 2 5)))
+;;       (inc (inc (inc (+ 1 5))))
+;;         (inc (inc (inc (inc (+ 0 5)))))
+;;           (inc (inc (inc (inc  5))))
+;;         (inc (inc (inc 6)))
+;;       (inc (inc 7))
+;;     (inc 8)
+;;   9
+;;
+;; Second process is iterative, the state can be maintained by only 'a' and 'b', without the need
+;; to know any previous recursive call states.
+;;
+;; The latter example with (+ 4 5) generates the calls:
+;; (+ 4 5)
+;; (+ 3 6)
+;; (+ 2 7)
+;; (+ 1 8)
+;; (+ 0 9)
+;; b -> (9)
+
+;; Exercise 1.10
+
+;; 1024
+;; 65536
+;; 65536
+;;
+;; Omitted mathematical stuff
+;;
+;; Exercise 1.11
+;;
+;;
+(define (f-recursive n)
+  (if (< n 3)
+      n
+      (+ (f (- n 1)) (* 2 (f (- n 2))) (* 3 (f (- n 3))))))
+
+(define (f-iterative n)
+  (define (iter count psum ppsum pppsum)
+    (if (> count n)
+	psum
+	(iter (+ count 1) (+ psum (* 2 ppsum) (* 3 pppsum)) psum ppsum)))
+  (iter 3 2 1 0))
```

Comment:
nit: for n<3, you need to return n directly, otherwise your f-iterative returns 2.